src/components/UserRegister.tsx

```diff
@@ -0,0 +1,278 @@
+import React, { useEffect, useState } from "react";
+import { useForm } from "react-hook-form";
+import { makeStyles, createStyles, Theme } from "@material-ui/core/styles";
+import LoadingOverlay from "react-loading-overlay-ts";
+import Grid from "@material-ui/core/Grid";
+import TextField from "@material-ui/core/TextField";
+import Australia from "/home/ishfaq/training-app/src/flags/Australia.png";
+import america from "/home/ishfaq/training-app/src/flags/america.png";
+import FlagSelect from "./FlagSelect";
+import Logo from "./Logo.png";
+import { IFormInput } from "./interfaces";
+
+import { useHistory } from "react-router-dom";
+
+const flags = {
+  imgs: [america, Australia],
+  name: ["+1-United States", "+61-Australia"],
+};
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      backgroundColor: "#FFFFFF",
+      maxWidth: "414px",
+      height: "896px",
+      marginTop: "30px",
+      paddingLeft: "25px",
+      "& .MuiGrid-spacing-xs-2 > .MuiGrid-item": {
+        padding: "18px",
+      },
+      "& .MuiFormLabel-root.Mui-focused": {
+        color: "#02E0B1",
+      },
+    },
+
+    logo: {
+      marginTop: "34px",
+      marginBottom: "30px",
+      marginLeft: "10px",
+      width: "96px",
+      height: "30px",
+    },
+
+    formnames: {
+      width: "170px",
+      height: "58px",
+      marginLeft: "10px",
+      marginBottom: "20px",
+    },
+    formfields: {
+      width: "366px",
+      height: "58px",
+    },
+    phonenumber: {
+      width: "366px",
+      height: "58px",
+
+      "& .MuiInputLabel-formControl": {
+        paddingLeft: "30px",
+      },
+      "& .MuiOutlinedInput-input": {
+        paddingLeft: "44px",
+      },
+      
+    },
+    container: {
+      position: "relative",
+      padding: "0",
+      margin: "0",
+
+      "& .MuiFormLabel-root.Mui-focused": {
+        paddingLeft: "25px",
+      },
+    },
+    flagimg: {
+      position: "absolute",
+      bottom: "20px",
+      left: "12px",
+      width: "24px",
+      height: "16px",
+    },
+  })
+);
+
+function capitalizeFirstLetter(string: string) {
```

Comment:
type and parameter name should be different, string is reserved key in react, you should not use that as a variable name